### PR TITLE
Some basic cleaning of security, kernel and web glue code

### PR DESCRIPTION
**Pull request by arjantijms**
Note that this is a simple pass over the code. There are still many more issues to address.

### CODE REVIEW

**Comment by avpinchuk** [comment]
`nucleus/core/kernel/src/test/java/com/sun/enterprise/v3/server/AppServerStartupTest.java` contains reference to the old `env` variable in several places:

```
[ERROR] /home/jenkins/agent/workspace/_test-using-jenkinsfile_PR-24425/nucleus/core/kernel/src/test/java/com/sun/enterprise/v3/server/AppServerStartupTest.java:[177,59] cannot find symbol
[ERROR]   symbol:   variable env
...
```

**Comment by arjantijms** [comment]
> [ERROR] /home/jenkins/agent/workspace/_test-using-jenkinsfile_PR-24425/nucleus/core/kernel/src/test/java/com/sun/enterprise/v3/server/AppServerStartupTest.java:[177,59] cannot find symbol
> [ERROR]   symbol:   variable env
> ...
> ```

My bad, didn't commit everything. Now I have ;)

File: appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardPipeline.java [hunk 1]
```diff
@@ -192,88 +179,77 @@ public void setContainer(Container container) {
...
+            }
         }
-        if ((basic != null) && (basic instanceof Lifecycle))
-            ((Lifecycle) basic).start();
 
-        /** CR 6411114 (MBean registration moved to ValveBase.start())
-        if( basic!=null )
-            registerValve(basic);
-        */
+        if ((basic != null) && (basic instanceof Lifecycle)) {
```
Comment (avpinchuk) [inline appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardPipeline.java:230]: Maybe remove `null` check?

File: appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardPipeline.java [hunk 2]
```diff
@@ -284,61 +260,51 @@ public synchronized void stop() throws LifecycleException {
         lifecycle.fireLifecycleEvent(STOP_EVENT, null);
 
         // Stop the Valves in our pipeline (including the basic), if any
-        if ((basic != null) && (basic instanceof Lifecycle))
+        if ((basic != null) && (basic instanceof Lifecycle)) {
```
Comment (avpinchuk) [inline appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardPipeline.java:263]: `null` check is redundant

File: appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardService.java [hunk 3]
```diff
@@ -591,77 +529,67 @@ public void stop() throws LifecycleException {
...
-    public void initialize()
-            throws LifecycleException
-    {
+    @Override
+    public void initialize() throws LifecycleException {
         // Service shouldn't be used with embedded, so it doesn't matter
         if (initialized) {
-            if (log.isLoggable(Level.INFO)) {
-                log.log(Level.INFO, LogFacade.SERVICE_HAS_BEEN_INIT);
+            if (log.isLoggable(INFO)) {
```
Comment (avpinchuk) [inline appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardService.java:542]: Is it needed here?

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/SetCommand.java [hunk 4]
```diff
@@ -388,18 +383,18 @@ public boolean execute(AdminCommandContext context) {
                 attrChanges.put(ctx.attrName, value);
             }
 
-            List<Map.Entry> mNodes = new ArrayList(matchingNodes.entrySet());
+            List<Map.Entry<Dom, String>> mNodes = new ArrayList(matchingNodes.entrySet());
```
Comment (avpinchuk) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/SetCommand.java:386]: Raw list creation

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/V2DottedNameSupport.java [hunk 5]
```diff
@@ -340,55 +341,67 @@ public Named[] getNamedNodes(List<? extends Named> target, List<? extends Named>
...
 
-    public List<Map.Entry> sortNodesByDottedName(Map<Dom, String> nodes) {
-        List<Map.Entry> mapEntries = new ArrayList(nodes.entrySet());
-        Collections.sort(mapEntries, new Comparator() {
-            public int compare(Object o1, Object o2) {
-                return (((String)(((Map.Entry) (o1)).getValue())).compareTo((String)(((Map.Entry) (o2)).getValue())));
+    public List<Entry<Dom, String>> sortNodesByDottedName(Map<Dom, String> nodes) {
+        List<Entry<Dom, String>> mapEntries = new ArrayList<>(nodes.entrySet());
+
+        Collections.sort(mapEntries, new Comparator<Entry<Dom, String>>() {
```
Comment (avpinchuk) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/V2DottedNameSupport.java:386]: MAybe use `List.sort()` instead?

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/V2DottedNameSupport.java [hunk 6]
```diff
@@ -340,55 +341,67 @@ public Named[] getNamedNodes(List<? extends Named> target, List<? extends Named>
...
     }
 
-    public List<org.glassfish.flashlight.datatree.TreeNode> sortTreeNodesByCompletePathName(List<org.glassfish.flashlight.datatree.TreeNode> nodes) {
-        Collections.sort(nodes, new Comparator() {
-            public int compare(Object o1, Object o2) {
-                return (((String)(((org.glassfish.flashlight.datatree.TreeNode) (o1)).getCompletePathName())).compareTo(
-                        (String)(((org.glassfish.flashlight.datatree.TreeNode) (o2)).getCompletePathName())));
+    public List<org.glassfish.flashlight.datatree.TreeNode> sortTreeNodesByCompletePathName(
+            List<org.glassfish.flashlight.datatree.TreeNode> nodes) {
+        Collections.sort(nodes, new Comparator<org.glassfish.flashlight.datatree.TreeNode>() {
```
Comment (avpinchuk) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/V2DottedNameSupport.java:398]: And here

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/services/impl/GrizzlyService.java [hunk 7]
```diff
@@ -83,61 +81,65 @@
...
+    private static final Logger LOGGER = KernelLoggerInfo.getLogger();
+    private static final String NETWORK_CONFIG_PREFIX = "";
 
-    @Inject
-    private ServiceLocator serviceLocator;
+    private final Set<MapperUpdateListener> mapperUpdateListeners = newSetFromMap(new ConcurrentHashMap<>());
 
-    @Inject
-    Transactions transactions;
+    private final Collection<NetworkProxy> proxies = new LinkedBlockingQueue<NetworkProxy>();
```
Comment (avpinchuk) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/services/impl/GrizzlyService.java:112]: Redundant type argument in `new LBQ()`

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/services/impl/GrizzlyService.java [hunk 8]
```diff
@@ -545,20 +534,20 @@ public List<Future<Result<Thread>>> getFutures() {
...
     /*
      * Creates a new NetworkProxy for a particular HttpListner
+     *
      * @param listener NetworkListener
+     *
      * @param networkConfig HttpService
      */
     public synchronized Future<Result<Thread>> createNetworkProxy(NetworkListener listener) {
-
         if (!Boolean.valueOf(listener.getEnabled())) {
```
Comment (avpinchuk) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/services/impl/GrizzlyService.java:543]: Maybe better `Boolean.parseBoolean()`?

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/services/impl/GrizzlyService.java [hunk 9]
```diff
@@ -796,28 +771,25 @@ private void registerAdapter(org.glassfish.api.container.Adapter a) throws Endpo
...
-                StringUtils.parseStringList(vsNetworkListeners, ",");
+            List<String> vsNetworkListenerList = StringUtils.parseStringList(vsNetworkListeners, ",");
             if (vsNetworkListenerList != null && !vsNetworkListenerList.isEmpty()) {
                 for (String vsNetworkListener : vsNetworkListenerList) {
                     for (NetworkListener networkListener : networkListenerList) {
-                        if (networkListener.getName().equals(vsNetworkListener) &&
-                            Boolean.valueOf(networkListener.getEnabled())) {
-                            addressInfos.add(new AddressInfo(networkListener.getAddress(),
-                                                             networkListener.getPort()));
+                        if (networkListener.getName().equals(vsNetworkListener) && Boolean.valueOf(networkListener.getEnabled())) {
```
Comment (avpinchuk) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/services/impl/GrizzlyService.java:791]: Use `Boolean.parseBoolean()` too

**Review by gurunrao — APPROVED** [review]
LGTM

**Comment by arjantijms** [comment]
@avpinchuk thanks for the comments. As mentioned, this is just a quick pass over the code. There are literally hundreds of additional things to fix.

**Pull request merged**